redux/actions.js

```diff
@@ -0,0 +1,37 @@
+import * as types from "./actionTypes";
+
+// Action Creators
+export function setCurrentUser(user) {
+    return {
+        type: types.SET_CURRENT_USER,
+        payload: user
+    };
+}
+
+export function setCurrentChat(chatId) {
+    return {
+        type: types.SET_CURRENT_CHAT,
+        payload: chatId
+    };
+}
+
+export function addMessage(id, cid, message, user, isSelf, emojis) {
+    return {
+        type: types.ADD_MESSAGE,
+        payload: { "id": id,
```

Comment:
just kinda an fyi, it's not something that's wrong, but in javascript if the key is the same as the value, you just need to write it once! like payload: { id, chats_id: cid, message, user }

Reply:
Good catch! You don't need quotes around the keys either.

Reply:
Since the payload is an object that you're defining here, I would follow this format:
```
payload: {
    id,
    chats_id: cid,
    ...
}
```

Reply:
Awesome, great points, changed accordingly.